File: 2D Array/java/sortedmatrix.java
import java.util.Arrays;

public class sortedmatrix {
    static final int N = 4;

    public static void printSortedMatrix(int[][] mat) {
        int size = N * N;
        int[] arr = new int[size];
        int k = 0;

        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++) {
                arr[k++] = mat[i][j];
            }
        }

        Arrays.sort(arr);

        for (int i = 0; i < size; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        int[][] mat = { {10, 20, 30, 40},
                        {15, 25, 35, 45},
                        {27, 29, 37, 48},
                        {32, 33, 39, 50} };

        System.out.print("Sorted elements: ");
        printSortedMatrix(mat);
    }
}

File: 2D Array/C/sortedmatrix.c
#include <stdio.h>
#include <stdlib.h>

#define N 4  

int compare(const void *a, const void *b) {
    return (*(int*)a - *(int*)b);
}

void printSortedMatrix(int mat[N][N]) {
    int size = N * N;
    int arr[size];
    int k = 0;

    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            arr[k++] = mat[i][j];
        }
    }

    qsort(arr, size, sizeof(int), compare);

    for (int i = 0; i < size; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");
}

int main() {
    int mat[N][N] = { {10, 20, 30, 40},
                      {15, 25, 35, 45},
                      {27, 29, 37, 48},
                      {32, 33, 39, 50} };

    printf("Sorted elements: ");
    printSortedMatrix(mat);

    return 0;
}

